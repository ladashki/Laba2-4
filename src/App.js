import React from 'react';
import Info from "./components/info";
import QuoteShow from "./components/quote";
//import './App.css';

class App extends React.Component {
    
   // state={ quote: undefined }

 constructor(){
    super();
    this.state = { quote: null };
}   
    
    
getQuote = async(e) => {
    e.preventDefault();
    const api_url = await fetch("https://api.kanye.rest")
    const data = await api_url.json();
    console.log(data);
    
    this.setState({
       quote:  data.quote
    });
}


    render(){
        return(
        <div> 
          <Info quoteMethod={this.getQuote} />          
           <QuoteShow quote={this.state.quote} />  
        </div>

        );
    }
}

export default App;


























