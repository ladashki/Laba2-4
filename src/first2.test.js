import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';

const testData = {
  data: {
"quote":"Tweeting is legal and also therapeutic"
           },
};

describe('Snapshot test', () => {
  test('Matches the snapshot', () => {
  
    const AppInfo = create(<App AppData={testData.data.runs} isLoading={false} />);
    expect(AppInfo.toJSON()).toMatchSnapshot();
  });
});


