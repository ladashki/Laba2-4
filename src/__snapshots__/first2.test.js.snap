// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Matches the snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <button>
       refresh 
    </button>
  </form>
  <div>
    <h3>
       
       
    </h3>
  </div>
</div>
`;
